app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js

```diff
@@ -262,12 +264,26 @@ ManageIQ.angular.app.controller('emsCommonFormController', ['$http', '$scope', '
       $scope.emsCommonModel.kubevirt_password               = data.kubevirt_password;
 
       miqService.sparkleOff();
-
+      $scope.hideDisabledTabs();
       $scope.afterGet  = true;
       $scope.modelCopy = angular.copy( $scope.emsCommonModel );
     }
   };
 
+  $scope.hideDisabledTabs = function() {
+    if ($scope.emsCommonModel.alerts_selection === 'disabled') {
+      $("#alerts_tab").hide();
+    }
+
+    if ($scope.emsCommonModel.metrics_selection === 'disabled') {
+      $("#metrics_tab").hide();
+    }
+
+    if ($scope.emsCommonModel.virtualization_selection === 'disabled') {
+      $("#virtualization_tab").hide();
+    }
+  }
```

Comment:
@masayag While this is working nice, I'm thinking, for consistency reasons should we be handling this in the haml itself? - since we do not handle any tab disabling for other provider types in the controller.

Although I also think that if it's going to make the haml look overly complicated, we should keep the tab hiding in the controller itself.

Can you move the `$scope.hideDisabledTabs();` to the `selectionChanged` methods, as shown below -
...that way we at least keep the tab hiding logic confined to the related UI event blocks in the controller
```diff
 $scope.metricSelectionChanged = function() {
     $scope.tabSelectionChanged("#metrics_tab", $scope.emsCommonModel.metrics_selection);
+    $scope.hideDisabledTabs();
   };
 
   $scope.alertsSelectionChanged = function() {
     $scope.tabSelectionChanged("#alerts_tab", $scope.emsCommonModel.alerts_selection);
+    $scope.hideDisabledTabs();
   };
 
   $scope.virtualizationSelectionChanged = function() {
     $scope.tabSelectionChanged("#virtualization_tab", $scope.emsCommonModel.virtualization_selection);
+    $scope.hideDisabledTabs();
   };
```

Reply:
@AparnaKarve moving the call of `$scope.hideDisabledTabs();` to `selectionChanged` won't fix the issue when editing an exiting provider or when adding a new one.

The partially visible tabs remains until a change to any of the existing tabs is done. That's the reason why the call to  `$scope.hideDisabledTabs();` was added to those specific functions.

![selection_188](https://user-images.githubusercontent.com/316242/35196049-9ab84892-fed5-11e7-915b-cce746a5c62e.png)

ATM, just updated the function to use `angular.element`.

Reply:
@masayag I noticed that the Alerts and the Metrics tab implementation has some issues. As in, those tabs do not display the error indicator (red exclamation mark) when there are Required /Validation Required errors on the form. So I think a separate PR needs to deal with all the issues related to Alerts and the Metrics tab.

Since the Virtualization tab changes in 31f9f0f2e290875e4340fdb5cfe2054b722a9022 are related to Alert/Metrics tabs, I think we should hold off on these changes for now.
Let's keep this PR strictly about kubevirt UI only.

Reply:
@AparnaKarve I've removed the metrics/alerts from the last commit.

Please let me know if you wish to remove the that commit completely or to leave it as in the latest PR version.

Reply:
@masayag let's remove the last commit completely.
(sorry for the trouble)

I do not have kubevirt credentials to test this - but I assume you / @zeari have already tested this. 

Once you have removed the last commit, I think we should be pretty close to merge.